spec.bs

```diff
@@ -439,13 +439,24 @@ A [=/request=] has an associated <dfn for=request>private token issuers</dfn>, w
 
 Note: [=request/Private token refresh policy=] is ignored unless [=request/private token operation=] is <code>"token-redemption"</code>. [=request/Private token issuers=] is ignored unless [=request/private token operation=] is <code>"send-redemption-record"</code>. [=request/Private token issuers=] must be specified and non-empty when [=request/private token operation=] is <code>"send-redemption-record"</code>.
 
+This specification defines two new [=policy-controlled features=]. Exactly one of these policy features applies for a given Private State Token operation.
+
+The [=policy-controlled feature=] identified by "<dfn data-dfn-for="policy-controlled feature"><code>private-state-token-issuance</code></dfn>" applies for the <code>"token-request"</code> operation. The [=default allowlist=] for this feature is <code>["self"]</code>.
+
+The [=policy-controlled feature=] identified by "<dfn data-dfn-for="policy-controlled feature"><code>private-state-token-redemption</code></dfn>" applies for the <code>"send-redemption-record"</code> and <code>"token-redemption"</code> operations. The [=default allowlist=] for this feature is <code>["self"]</code>.
+
 A [=request=] has an associated <dfn for="request">pstPretokens</dfn>, which is null or a [=byte sequence=].
 
 Add the following steps to the <code><a constructor lt="Request()">new Request (<var ignore>input</var>, |init|)</a></code> constructor, before step 28 ("<code>Set [=this=]'s [=Request/request=] to |request|</code>"):
 
  1. If |init|["{{RequestInit/privateToken}}"] [=map/exists=]:
      1. Set <var ignore>request</var>'s [=request/private token operation=]</a> to |init|["{{RequestInit/privateToken}}"]["{{PrivateToken/operation}}"].
-     1. If |init|["{{RequestInit/privateToken}}"]["{{PrivateToken/operation}}"] is <code>"token-request"</code>, then abort these steps.
+     1. If |init|["{{RequestInit/privateToken}}"]["{{PrivateToken/operation}}"] is <code>"token-request"</code>:
+         1. If [$Should request be allowed to use feature?|should request be allowed$] for "<code>[=policy-controlled feature/private-state-token-issuance=]</code>" returns false, then throw a "{{NotAllowedError}}" {{DOMException}}.
+         1. Abort the remaining steps.
```

Comment:
This will skip the rest of the algorithm for any call with init.privateToken.operation=="token-request", not only if permission policy disallows the call.

Strictly, you don't even need to say to abort the steps after you throw: https://infra.spec.whatwg.org/#algorithm-control-flow and https://webidl.spec.whatwg.org/#es-handling-exceptions say that throwing implies skipping the remaining steps. So you can just delete this line to fix the problem.

Reply:
Algorithm is done if operation is `"token-request"`. Skipping rest of the algorithm is the intended behavior.

Reply:
Since this is a monkeypatch to the `new Request` algorithm in Fetch, aborting here means that it'll skip the rest of that constructor ... and not return anything as the new Request object. If the control flow you want is to skip the rest of the contents of the "If" block above, the clearest way to do that is probably to extract the body into a named algorithm. That'll also make merging the monkeypatch easier.